boards/qemu_rv32_virt/Makefile

```diff
@@ -0,0 +1,60 @@
+# Makefile for building the Tock kernel for the qemu-system-riscv32
+# virt machine type
+
+TARGET=riscv32imac-unknown-none-elf
+PLATFORM=qemu_rv32_virt
+
+include ../Makefile.common
+
+WORKING_QEMU_VERSION=5.2.0
+
+# Run the kernel inside a qemu-riscv32-system "virt" machine type
+# simulation
+#
+# Peripherals attached by default:
+# - 16550 UART (attached to stdio by default)
+# - VirtIO EntropySource (default backend /dev/random)
+#
+# By default a VirtIO NetworkCard is _not_ attached, since creating a
+# TAP device on the host will require root or further system
+# configuration. The configuration options to enable the network card
+# are included as comments.
+#
+# Requires that a qemu-riscv32-system binary is in the user's
+# PATH. The tested & verified QEMU version is printed along with the
+# one used. No actual version check is performed given the simulation
+# might work with different version, though should at leat work on the
+# tested one.
+run: $(TOCK_ROOT_DIRECTORY)target/$(TARGET)/release/$(PLATFORM).bin
+	@echo
+	@echo -e "Running $$(qemu-system-riscv32 --version | head -n1)" \
+	  "(tested: $(WORKING_QEMU_VERSION)) with\n  - kernel $^"
+	@echo "To exit type C-a x"
+	@echo
+	qemu-system-riscv32 \
+	  -machine virt \
+	  -bios $^ \
+	  -global virtio-mmio.force-legacy=false \
+	  -device virtio-rng-device \
+	  -nographic
+	  @# attaching a TAP network device requires proper permissions
+	  @# to create a tuntap device or access to an existing device
+	  @# -netdev tap,id=n0,script=no,downscript=no
+	  @# -device virtio-net-device,netdev=n0
```

Comment:
We can do networking without TAP, but let's leave that for the future

Reply:
Right, for outgoing connections that works fine (using QEMU's userspace networking functionality, right?) but for incoming connections it wouldn't. My dummy/demo usecase for this is going to be running a webserver, which is great because you have some static code running in the QEMU machine and can interact from within your own OS / browser. This is why I decided to go with TAP at first, in addition to it being easier to debug.

Reply:
You can setup port forwarding with the userspace networking. So you should still be able to host a webserver (ICMP doesn't work though)

Reply:
That's good to know. Given this doesn't add any networking in this PR, I'd say we change the comment once VirtIO is introduced proper.